.github/workflows/ci-e2e.yaml

```diff
@@ -19,27 +19,20 @@ on:
       - 'LICENSE'
 
 jobs:
-  event_payload:
+  e2e_test_suite:
+    name: E2E Test Suite
     runs-on: ubuntu-latest
-    steps:
-      - run: cat $GITHUB_EVENT_PATH
-  authorize:
     # see https://docs.github.com/en/graphql/reference/enums#commentauthorassociation
     # for info on author association
     environment:
       ${{ github.event_name == 'pull_request_target' &&
       github.event.pull_request.author_association != 'OWNER' &&
       github.event.pull_request.author_association != 'MEMBER' &&
       github.event.pull_request.author_association != 'COLLABORATOR' &&
```

Comment:
Bit confused by this. Does an Owner have all of these or just one? Should this be an OR? I'm not seeing what fixes the issue i was seeing since I'm an OWNER of the repo currently.

Reply:
A user only needs to be one of those. The problem I detected when we first merged this authorize thing is that in the `pull_request_target` events we are always being marked as `CONTRIBUTORS`, even if we already are `MEMBERS` or `OWNERS`. I'll try to look for a better check as `CONTRIBUTOR` is a little bit too open as it includes anyone that has ever made a commit to the repo. For now this should avoid us from having to approve the e2e workflow for our own PRs.

An example payload for onw of your PRs (search for the 'author_association' field):

```json
{
  "action": "synchronize",
  "after": "837c57593220ca3fac50a391dec8549106940ab4",
  "before": "995a0c20025bc2e4b21f9a1819486199b755a435",
  "number": 312,
  "organization": {
    "avatar_url": "https://avatars.githubusercontent.com/u/78971243?v=4",
    "description": "Gateway Policies for Kubernetes",
    "events_url": "https://api.github.com/orgs/Kuadrant/events",
    "hooks_url": "https://api.github.com/orgs/Kuadrant/hooks",
    "id": 78971243,
    "issues_url": "https://api.github.com/orgs/Kuadrant/issues",
    "login": "Kuadrant",
    "members_url": "https://api.github.com/orgs/Kuadrant/members{/member}",
    "node_id": "MDEyOk9yZ2FuaXphdGlvbjc4OTcxMjQz",
    "public_members_url": "https://api.github.com/orgs/Kuadrant/public_members{/member}",
    "repos_url": "https://api.github.com/orgs/Kuadrant/repos",
    "url": "https://api.github.com/orgs/Kuadrant"
  },
  "pull_request": {
    "_links": {
      "comments": {
        "href": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/issues/312/comments"
      },
      "commits": {
        "href": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/pulls/312/commits"
      },
      "html": {
        "href": "https://github.com/Kuadrant/multicluster-gateway-controller/pull/312"
      },
      "issue": {
        "href": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/issues/312"
      },
      "review_comment": {
        "href": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/pulls/comments{/number}"
      },
      "review_comments": {
        "href": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/pulls/312/comments"
      },
      "self": {
        "href": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/pulls/312"
      },
      "statuses": {
        "href": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/statuses/837c57593220ca3fac50a391dec8549106940ab4"
      }
    },
    "active_lock_reason": null,
    "additions": 594,
    "assignee": null,
    "assignees": [],
    "author_association": "CONTRIBUTOR",
    "auto_merge": null,
    "base": {
      "label": "Kuadrant:main",
      "ref": "main",
      "repo": {
        "allow_auto_merge": false,
        "allow_forking": true,
        "allow_merge_commit": true,
        "allow_rebase_merge": true,
        "allow_squash_merge": true,
        "allow_update_branch": false,
        "archive_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/{archive_format}{/ref}",
        "archived": false,
        "assignees_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/assignees{/user}",
        "blobs_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/git/blobs{/sha}",
        "branches_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/branches{/branch}",
        "clone_url": "https://github.com/Kuadrant/multicluster-gateway-controller.git",
        "collaborators_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/collaborators{/collaborator}",
        "comments_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/comments{/number}",
        "commits_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/commits{/sha}",
        "compare_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/compare/{base}...{head}",
        "contents_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/contents/{+path}",
        "contributors_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/contributors",
        "created_at": "2022-12-15T12:17:57Z",
        "default_branch": "main",
        "delete_branch_on_merge": false,
        "deployments_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/deployments",
        "description": "multi-cluster gateway controller, manages multi-cluster gateways based on gateway api and policy attachment",
        "disabled": false,
        "downloads_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/downloads",
        "events_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/events",
        "fork": false,
        "forks": 12,
        "forks_count": 12,
        "forks_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/forks",
        "full_name": "Kuadrant/multicluster-gateway-controller",
        "git_commits_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/git/commits{/sha}",
        "git_refs_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/git/refs{/sha}",
        "git_tags_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/git/tags{/sha}",
        "git_url": "git://github.com/Kuadrant/multicluster-gateway-controller.git",
        "has_discussions": true,
        "has_downloads": true,
        "has_issues": true,
        "has_pages": false,
        "has_projects": true,
        "has_wiki": true,
        "homepage": "",
        "hooks_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/hooks",
        "html_url": "https://github.com/Kuadrant/multicluster-gateway-controller",
        "id": 578589674,
        "is_template": false,
        "issue_comment_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/issues/comments{/number}",
        "issue_events_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/issues/events{/number}",
        "issues_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/issues{/number}",
        "keys_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/keys{/key_id}",
        "labels_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/labels{/name}",
        "language": "Go",
        "languages_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/languages",
        "license": {
          "key": "apache-2.0",
          "name": "Apache License 2.0",
          "node_id": "MDc6TGljZW5zZTI=",
          "spdx_id": "Apache-2.0",
          "url": "https://api.github.com/licenses/apache-2.0"
        },
        "merge_commit_message": "PR_TITLE",
        "merge_commit_title": "MERGE_MESSAGE",
        "merges_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/merges",
        "milestones_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/milestones{/number}",
        "mirror_url": null,
        "name": "multicluster-gateway-controller",
        "node_id": "R_kgDOInyT6g",
        "notifications_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/notifications{?since,all,participating}",
        "open_issues": 71,
        "open_issues_count": 71,
        "owner": {
          "avatar_url": "https://avatars.githubusercontent.com/u/78971243?v=4",
          "events_url": "https://api.github.com/users/Kuadrant/events{/privacy}",
          "followers_url": "https://api.github.com/users/Kuadrant/followers",
          "following_url": "https://api.github.com/users/Kuadrant/following{/other_user}",
          "gists_url": "https://api.github.com/users/Kuadrant/gists{/gist_id}",
          "gravatar_id": "",
          "html_url": "https://github.com/Kuadrant",
          "id": 78971243,
          "login": "Kuadrant",
          "node_id": "MDEyOk9yZ2FuaXphdGlvbjc4OTcxMjQz",
          "organizations_url": "https://api.github.com/users/Kuadrant/orgs",
          "received_events_url": "https://api.github.com/users/Kuadrant/received_events",
          "repos_url": "https://api.github.com/users/Kuadrant/repos",
          "site_admin": false,
          "starred_url": "https://api.github.com/users/Kuadrant/starred{/owner}{/repo}",
          "subscriptions_url": "https://api.github.com/users/Kuadrant/subscriptions",
          "type": "Organization",
          "url": "https://api.github.com/users/Kuadrant"
        },
        "private": false,
        "pulls_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/pulls{/number}",
        "pushed_at": "2023-07-10T08:07:07Z",
        "releases_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/releases{/id}",
        "size": 61912,
        "squash_merge_commit_message": "COMMIT_MESSAGES",
        "squash_merge_commit_title": "COMMIT_OR_PR_TITLE",
        "ssh_url": "git@github.com:Kuadrant/multicluster-gateway-controller.git",
        "stargazers_count": 4,
        "stargazers_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/stargazers",
        "statuses_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/statuses/{sha}",
        "subscribers_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/subscribers",
        "subscription_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/subscription",
        "svn_url": "https://github.com/Kuadrant/multicluster-gateway-controller",
        "tags_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/tags",
        "teams_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/teams",
        "topics": [
          "gateway-api",
          "global-load-balancing",
          "multi-cloud",
          "multi-cluster"
        ],
        "trees_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/git/trees{/sha}",
        "updated_at": "2023-07-07T17:33:26Z",
        "url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller",
        "use_squash_pr_title_as_default": false,
        "visibility": "public",
        "watchers": 4,
        "watchers_count": 4,
        "web_commit_signoff_required": false
      },
      "sha": "f363458f9d06ec63b6c1f97769a59c3e2cc913da",
      "user": {
        "avatar_url": "https://avatars.githubusercontent.com/u/78971243?v=4",
        "events_url": "https://api.github.com/users/Kuadrant/events{/privacy}",
        "followers_url": "https://api.github.com/users/Kuadrant/followers",
        "following_url": "https://api.github.com/users/Kuadrant/following{/other_user}",
        "gists_url": "https://api.github.com/users/Kuadrant/gists{/gist_id}",
        "gravatar_id": "",
        "html_url": "https://github.com/Kuadrant",
        "id": 78971243,
        "login": "Kuadrant",
        "node_id": "MDEyOk9yZ2FuaXphdGlvbjc4OTcxMjQz",
        "organizations_url": "https://api.github.com/users/Kuadrant/orgs",
        "received_events_url": "https://api.github.com/users/Kuadrant/received_events",
        "repos_url": "https://api.github.com/users/Kuadrant/repos",
        "site_admin": false,
        "starred_url": "https://api.github.com/users/Kuadrant/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/Kuadrant/subscriptions",
        "type": "Organization",
        "url": "https://api.github.com/users/Kuadrant"
      }
    },
    "body": "Adds an option to the `DNSPolicy.LoadBalancing.Weighted` spec to add custom weights for attributes set on target clusters via labels. The custom attributes are checked during the calculation of the target clusters weight, and if attributes match, the custom weight is applied.\r\n\r\nExample\r\n\r\n```\r\nspec:\r\n  loadBalancing:\r\n    weighted:\r\n      custom:\r\n      - value: AWS\r\n        weight: 60\r\n      - value: GCP\r\n        weight: 255\r\n      defaultWeight: 120\r\n```\r\n```\r\nkubectl label --overwrite managedcluster cluster-1 kuadrant.io/lb-attribute-custom-weight=AWS\r\nkubectl label --overwrite managedcluster cluster-2 kuadrant.io/lb-attribute-custom-weight=GCP\r\nkubectl label --overwrite managedcluster cluster-3 kuadrant.io/lb-attribute-custom-weight-\r\n```\r\n\r\nThe above would favour the GCP cluster(cluster-2) over any other, and the AWS cluster(cluster-1) would recieve the least traffic with cluster-3 falling back to the default weight getting a medium amount of traffic.\r\n\r\ncloses #284 \r\n",
    "changed_files": 8,
    "closed_at": null,
    "comments": 1,
    "comments_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/issues/312/comments",
    "commits": 2,
    "commits_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/pulls/312/commits",
    "created_at": "2023-07-07T08:30:08Z",
    "deletions": 29,
    "diff_url": "https://github.com/Kuadrant/multicluster-gateway-controller/pull/312.diff",
    "draft": false,
    "head": {
      "label": "mikenairn:284-custom_weights",
      "ref": "284-custom_weights",
      "repo": {
        "allow_auto_merge": false,
        "allow_forking": true,
        "allow_merge_commit": true,
        "allow_rebase_merge": true,
        "allow_squash_merge": true,
        "allow_update_branch": false,
        "archive_url": "https://api.github.com/repos/mikenairn/multi-cluster-traffic-controller/{archive_format}{/ref}",
        "archived": false,
        "assignees_url": "https://api.github.com/repos/mikenairn/multi-cluster-traffic-controller/assignees{/user}",
        "blobs_url": "https://api.github.com/repos/mikenairn/multi-cluster-traffic-controller/git/blobs{/sha}",
        "branches_url": "https://api.github.com/repos/mikenairn/multi-cluster-traffic-controller/branches{/branch}",
        "clone_url": "https://github.com/mikenairn/multi-cluster-traffic-controller.git",
        "collaborators_url": "https://api.github.com/repos/mikenairn/multi-cluster-traffic-controller/collaborators{/collaborator}",
        "comments_url": "https://api.github.com/repos/mikenairn/multi-cluster-traffic-controller/comments{/number}",
        "commits_url": "https://api.github.com/repos/mikenairn/multi-cluster-traffic-controller/commits{/sha}",
        "compare_url": "https://api.github.com/repos/mikenairn/multi-cluster-traffic-controller/compare/{base}...{head}",
        "contents_url": "https://api.github.com/repos/mikenairn/multi-cluster-traffic-controller/contents/{+path}",
        "contributors_url": "https://api.github.com/repos/mikenairn/multi-cluster-traffic-controller/contributors",
        "created_at": "2022-12-16T14:56:47Z",
        "default_branch": "main",
        "delete_branch_on_merge": false,
        "deployments_url": "https://api.github.com/repos/mikenairn/multi-cluster-traffic-controller/deployments",
        "description": null,
        "disabled": false,
        "downloads_url": "https://api.github.com/repos/mikenairn/multi-cluster-traffic-controller/downloads",
        "events_url": "https://api.github.com/repos/mikenairn/multi-cluster-traffic-controller/events",
        "fork": true,
        "forks": 0,
        "forks_count": 0,
        "forks_url": "https://api.github.com/repos/mikenairn/multi-cluster-traffic-controller/forks",
        "full_name": "mikenairn/multi-cluster-traffic-controller",
        "git_commits_url": "https://api.github.com/repos/mikenairn/multi-cluster-traffic-controller/git/commits{/sha}",
        "git_refs_url": "https://api.github.com/repos/mikenairn/multi-cluster-traffic-controller/git/refs{/sha}",
        "git_tags_url": "https://api.github.com/repos/mikenairn/multi-cluster-traffic-controller/git/tags{/sha}",
        "git_url": "git://github.com/mikenairn/multi-cluster-traffic-controller.git",
        "has_discussions": false,
        "has_downloads": true,
        "has_issues": false,
        "has_pages": false,
        "has_projects": true,
        "has_wiki": true,
        "homepage": null,
        "hooks_url": "https://api.github.com/repos/mikenairn/multi-cluster-traffic-controller/hooks",
        "html_url": "https://github.com/mikenairn/multi-cluster-traffic-controller",
        "id": 579058927,
        "is_template": false,
        "issue_comment_url": "https://api.github.com/repos/mikenairn/multi-cluster-traffic-controller/issues/comments{/number}",
        "issue_events_url": "https://api.github.com/repos/mikenairn/multi-cluster-traffic-controller/issues/events{/number}",
        "issues_url": "https://api.github.com/repos/mikenairn/multi-cluster-traffic-controller/issues{/number}",
        "keys_url": "https://api.github.com/repos/mikenairn/multi-cluster-traffic-controller/keys{/key_id}",
        "labels_url": "https://api.github.com/repos/mikenairn/multi-cluster-traffic-controller/labels{/name}",
        "language": "Go",
        "languages_url": "https://api.github.com/repos/mikenairn/multi-cluster-traffic-controller/languages",
        "license": {
          "key": "apache-2.0",
          "name": "Apache License 2.0",
          "node_id": "MDc6TGljZW5zZTI=",
          "spdx_id": "Apache-2.0",
          "url": "https://api.github.com/licenses/apache-2.0"
        },
        "merge_commit_message": "PR_TITLE",
        "merge_commit_title": "MERGE_MESSAGE",
        "merges_url": "https://api.github.com/repos/mikenairn/multi-cluster-traffic-controller/merges",
        "milestones_url": "https://api.github.com/repos/mikenairn/multi-cluster-traffic-controller/milestones{/number}",
        "mirror_url": null,
        "name": "multi-cluster-traffic-controller",
        "node_id": "R_kgDOIoO87w",
        "notifications_url": "https://api.github.com/repos/mikenairn/multi-cluster-traffic-controller/notifications{?since,all,participating}",
        "open_issues": 0,
        "open_issues_count": 0,
        "owner": {
          "avatar_url": "https://avatars.githubusercontent.com/u/327352?v=4",
          "events_url": "https://api.github.com/users/mikenairn/events{/privacy}",
          "followers_url": "https://api.github.com/users/mikenairn/followers",
          "following_url": "https://api.github.com/users/mikenairn/following{/other_user}",
          "gists_url": "https://api.github.com/users/mikenairn/gists{/gist_id}",
          "gravatar_id": "",
          "html_url": "https://github.com/mikenairn",
          "id": 327352,
          "login": "mikenairn",
          "node_id": "MDQ6VXNlcjMyNzM1Mg==",
          "organizations_url": "https://api.github.com/users/mikenairn/orgs",
          "received_events_url": "https://api.github.com/users/mikenairn/received_events",
          "repos_url": "https://api.github.com/users/mikenairn/repos",
          "site_admin": false,
          "starred_url": "https://api.github.com/users/mikenairn/starred{/owner}{/repo}",
          "subscriptions_url": "https://api.github.com/users/mikenairn/subscriptions",
          "type": "User",
          "url": "https://api.github.com/users/mikenairn"
        },
        "private": false,
        "pulls_url": "https://api.github.com/repos/mikenairn/multi-cluster-traffic-controller/pulls{/number}",
        "pushed_at": "2023-07-10T08:12:00Z",
        "releases_url": "https://api.github.com/repos/mikenairn/multi-cluster-traffic-controller/releases{/id}",
        "size": 61946,
        "squash_merge_commit_message": "COMMIT_MESSAGES",
        "squash_merge_commit_title": "COMMIT_OR_PR_TITLE",
        "ssh_url": "git@github.com:mikenairn/multi-cluster-traffic-controller.git",
        "stargazers_count": 0,
        "stargazers_url": "https://api.github.com/repos/mikenairn/multi-cluster-traffic-controller/stargazers",
        "statuses_url": "https://api.github.com/repos/mikenairn/multi-cluster-traffic-controller/statuses/{sha}",
        "subscribers_url": "https://api.github.com/repos/mikenairn/multi-cluster-traffic-controller/subscribers",
        "subscription_url": "https://api.github.com/repos/mikenairn/multi-cluster-traffic-controller/subscription",
        "svn_url": "https://github.com/mikenairn/multi-cluster-traffic-controller",
        "tags_url": "https://api.github.com/repos/mikenairn/multi-cluster-traffic-controller/tags",
        "teams_url": "https://api.github.com/repos/mikenairn/multi-cluster-traffic-controller/teams",
        "topics": [],
        "trees_url": "https://api.github.com/repos/mikenairn/multi-cluster-traffic-controller/git/trees{/sha}",
        "updated_at": "2023-01-31T22:12:55Z",
        "url": "https://api.github.com/repos/mikenairn/multi-cluster-traffic-controller",
        "use_squash_pr_title_as_default": false,
        "visibility": "public",
        "watchers": 0,
        "watchers_count": 0,
        "web_commit_signoff_required": false
      },
      "sha": "837c57593220ca3fac50a391dec8549106940ab4",
      "user": {
        "avatar_url": "https://avatars.githubusercontent.com/u/327352?v=4",
        "events_url": "https://api.github.com/users/mikenairn/events{/privacy}",
        "followers_url": "https://api.github.com/users/mikenairn/followers",
        "following_url": "https://api.github.com/users/mikenairn/following{/other_user}",
        "gists_url": "https://api.github.com/users/mikenairn/gists{/gist_id}",
        "gravatar_id": "",
        "html_url": "https://github.com/mikenairn",
        "id": 327352,
        "login": "mikenairn",
        "node_id": "MDQ6VXNlcjMyNzM1Mg==",
        "organizations_url": "https://api.github.com/users/mikenairn/orgs",
        "received_events_url": "https://api.github.com/users/mikenairn/received_events",
        "repos_url": "https://api.github.com/users/mikenairn/repos",
        "site_admin": false,
        "starred_url": "https://api.github.com/users/mikenairn/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/mikenairn/subscriptions",
        "type": "User",
        "url": "https://api.github.com/users/mikenairn"
      }
    },
    "html_url": "https://github.com/Kuadrant/multicluster-gateway-controller/pull/312",
    "id": 1424239502,
    "issue_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/issues/312",
    "labels": [
      {
        "color": "ededed",
        "default": false,
        "description": null,
        "id": 5403347710,
        "name": "approved",
        "node_id": "LA_kwDOInyT6s8AAAABQhCK_g",
        "url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/labels/approved"
      }
    ],
    "locked": false,
    "maintainer_can_modify": true,
    "merge_commit_sha": "f33e0bbb5fe992776e009ccc83d7fa8d70de051c",
    "mergeable": null,
    "mergeable_state": "unknown",
    "merged": false,
    "merged_at": null,
    "merged_by": null,
    "milestone": null,
    "node_id": "PR_kwDOInyT6s5U5CuO",
    "number": 312,
    "patch_url": "https://github.com/Kuadrant/multicluster-gateway-controller/pull/312.patch",
    "rebaseable": null,
    "requested_reviewers": [
      {
        "avatar_url": "https://avatars.githubusercontent.com/u/37799801?v=4",
        "events_url": "https://api.github.com/users/makslion/events{/privacy}",
        "followers_url": "https://api.github.com/users/makslion/followers",
        "following_url": "https://api.github.com/users/makslion/following{/other_user}",
        "gists_url": "https://api.github.com/users/makslion/gists{/gist_id}",
        "gravatar_id": "",
        "html_url": "https://github.com/makslion",
        "id": 37799801,
        "login": "makslion",
        "node_id": "MDQ6VXNlcjM3Nzk5ODAx",
        "organizations_url": "https://api.github.com/users/makslion/orgs",
        "received_events_url": "https://api.github.com/users/makslion/received_events",
        "repos_url": "https://api.github.com/users/makslion/repos",
        "site_admin": false,
        "starred_url": "https://api.github.com/users/makslion/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/makslion/subscriptions",
        "type": "User",
        "url": "https://api.github.com/users/makslion"
      }
    ],
    "requested_teams": [],
    "review_comment_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/pulls/comments{/number}",
    "review_comments": 1,
    "review_comments_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/pulls/312/comments",
    "state": "open",
    "statuses_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/statuses/837c57593220ca3fac50a391dec8549106940ab4",
    "title": "Add custom weights to DNSPolicy",
    "updated_at": "2023-07-10T08:12:02Z",
    "url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/pulls/312",
    "user": {
      "avatar_url": "https://avatars.githubusercontent.com/u/327352?v=4",
      "events_url": "https://api.github.com/users/mikenairn/events{/privacy}",
      "followers_url": "https://api.github.com/users/mikenairn/followers",
      "following_url": "https://api.github.com/users/mikenairn/following{/other_user}",
      "gists_url": "https://api.github.com/users/mikenairn/gists{/gist_id}",
      "gravatar_id": "",
      "html_url": "https://github.com/mikenairn",
      "id": 327352,
      "login": "mikenairn",
      "node_id": "MDQ6VXNlcjMyNzM1Mg==",
      "organizations_url": "https://api.github.com/users/mikenairn/orgs",
      "received_events_url": "https://api.github.com/users/mikenairn/received_events",
      "repos_url": "https://api.github.com/users/mikenairn/repos",
      "site_admin": false,
      "starred_url": "https://api.github.com/users/mikenairn/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/mikenairn/subscriptions",
      "type": "User",
      "url": "https://api.github.com/users/mikenairn"
    }
  },
  "repository": {
    "allow_forking": true,
    "archive_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/{archive_format}{/ref}",
    "archived": false,
    "assignees_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/assignees{/user}",
    "blobs_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/git/blobs{/sha}",
    "branches_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/branches{/branch}",
    "clone_url": "https://github.com/Kuadrant/multicluster-gateway-controller.git",
    "collaborators_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/collaborators{/collaborator}",
    "comments_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/comments{/number}",
    "commits_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/commits{/sha}",
    "compare_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/compare/{base}...{head}",
    "contents_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/contents/{+path}",
    "contributors_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/contributors",
    "created_at": "2022-12-15T12:17:57Z",
    "default_branch": "main",
    "deployments_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/deployments",
    "description": "multi-cluster gateway controller, manages multi-cluster gateways based on gateway api and policy attachment",
    "disabled": false,
    "downloads_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/downloads",
    "events_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/events",
    "fork": false,
    "forks": 12,
    "forks_count": 12,
    "forks_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/forks",
    "full_name": "Kuadrant/multicluster-gateway-controller",
    "git_commits_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/git/commits{/sha}",
    "git_refs_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/git/refs{/sha}",
    "git_tags_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/git/tags{/sha}",
    "git_url": "git://github.com/Kuadrant/multicluster-gateway-controller.git",
    "has_discussions": true,
    "has_downloads": true,
    "has_issues": true,
    "has_pages": false,
    "has_projects": true,
    "has_wiki": true,
    "homepage": "",
    "hooks_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/hooks",
    "html_url": "https://github.com/Kuadrant/multicluster-gateway-controller",
    "id": 578589674,
    "is_template": false,
    "issue_comment_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/issues/comments{/number}",
    "issue_events_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/issues/events{/number}",
    "issues_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/issues{/number}",
    "keys_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/keys{/key_id}",
    "labels_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/labels{/name}",
    "language": "Go",
    "languages_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/languages",
    "license": {
      "key": "apache-2.0",
      "name": "Apache License 2.0",
      "node_id": "MDc6TGljZW5zZTI=",
      "spdx_id": "Apache-2.0",
      "url": "https://api.github.com/licenses/apache-2.0"
    },
    "merges_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/merges",
    "milestones_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/milestones{/number}",
    "mirror_url": null,
    "name": "multicluster-gateway-controller",
    "node_id": "R_kgDOInyT6g",
    "notifications_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/notifications{?since,all,participating}",
    "open_issues": 71,
    "open_issues_count": 71,
    "owner": {
      "avatar_url": "https://avatars.githubusercontent.com/u/78971243?v=4",
      "events_url": "https://api.github.com/users/Kuadrant/events{/privacy}",
      "followers_url": "https://api.github.com/users/Kuadrant/followers",
      "following_url": "https://api.github.com/users/Kuadrant/following{/other_user}",
      "gists_url": "https://api.github.com/users/Kuadrant/gists{/gist_id}",
      "gravatar_id": "",
      "html_url": "https://github.com/Kuadrant",
      "id": 78971243,
      "login": "Kuadrant",
      "node_id": "MDEyOk9yZ2FuaXphdGlvbjc4OTcxMjQz",
      "organizations_url": "https://api.github.com/users/Kuadrant/orgs",
      "received_events_url": "https://api.github.com/users/Kuadrant/received_events",
      "repos_url": "https://api.github.com/users/Kuadrant/repos",
      "site_admin": false,
      "starred_url": "https://api.github.com/users/Kuadrant/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/Kuadrant/subscriptions",
      "type": "Organization",
      "url": "https://api.github.com/users/Kuadrant"
    },
    "private": false,
    "pulls_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/pulls{/number}",
    "pushed_at": "2023-07-10T08:07:07Z",
    "releases_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/releases{/id}",
    "size": 61912,
    "ssh_url": "git@github.com:Kuadrant/multicluster-gateway-controller.git",
    "stargazers_count": 4,
    "stargazers_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/stargazers",
    "statuses_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/statuses/{sha}",
    "subscribers_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/subscribers",
    "subscription_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/subscription",
    "svn_url": "https://github.com/Kuadrant/multicluster-gateway-controller",
    "tags_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/tags",
    "teams_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/teams",
    "topics": [
      "gateway-api",
      "global-load-balancing",
      "multi-cloud",
      "multi-cluster"
    ],
    "trees_url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller/git/trees{/sha}",
    "updated_at": "2023-07-07T17:33:26Z",
    "url": "https://api.github.com/repos/Kuadrant/multicluster-gateway-controller",
    "visibility": "public",
    "watchers": 4,
    "watchers_count": 4,
    "web_commit_signoff_required": false
  },
  "sender": {
    "avatar_url": "https://avatars.githubusercontent.com/u/327352?v=4",
    "events_url": "https://api.github.com/users/mikenairn/events{/privacy}",
    "followers_url": "https://api.github.com/users/mikenairn/followers",
    "following_url": "https://api.github.com/users/mikenairn/following{/other_user}",
    "gists_url": "https://api.github.com/users/mikenairn/gists{/gist_id}",
    "gravatar_id": "",
    "html_url": "https://github.com/mikenairn",
    "id": 327352,
    "login": "mikenairn",
    "node_id": "MDQ6VXNlcjMyNzM1Mg==",
    "organizations_url": "https://api.github.com/users/mikenairn/orgs",
    "received_events_url": "https://api.github.com/users/mikenairn/received_events",
    "repos_url": "https://api.github.com/users/mikenairn/repos",
    "site_admin": false,
    "starred_url": "https://api.github.com/users/mikenairn/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/mikenairn/subscriptions",
    "type": "User",
    "url": "https://api.github.com/users/mikenairn"
  }
}
```

Reply:
> ``. I'll try to look for a better check as `CONTRIBUTOR` is a little bit too open as it includes anyone that has ever made a commit to the repo.

I see, yes this is no ideal.

I think also i didn't see we were using "!=", so if it's not any of these things, use this.

Generally I'm not convinced about using these environments. Unless there is a better way of managing these things we coudl consider using openshift ci for these e2e tests.